Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +22,72 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(current_ < 0){
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double totalHoursRemaining, msRemaining;
+        int h, m, s, ms;
+
+        if(isCharging())
+            totalHoursRemaining = totalAmpHoursUsed() / current_ * -1;
+        else
+            totalHoursRemaining = (BATTERY_AMP_HOUR_CAPACITY - totalAmpHoursUsed()) / current_;
+
+        while(totalHoursRemaining > 24){
+            totalHoursRemaining -= 24;
+        }
+        //Converting totalHoursRemaining into h, m, s, ms
+        h = totalHoursRemaining;
+
+        msRemaining = (totalHoursRemaining - h) * 3600000;
+        m = msRemaining / 60000;
+        msRemaining = msRemaining - (m * 60000);
+        s = msRemaining / 1000;
+        msRemaining = msRemaining - (s * 1000);
+        ms = msRemaining;
+
+        QTime timeRemaining(h, m, s, ms);
+        return timeRemaining;
+
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
+    double initialAmphoursUsed = (initialStateOfChargePercent_ * BATTERY_AMP_HOUR_CAPACITY)/100;
+    Q_UNUSED(batteryData)
     // This is where you can update your variables
     // Hint: There are many different ways that the totalAmpHoursUsed can be updated
     // i.e: Taking a running average of your data values, using most recent data points, etc.
+
+    oldCurrent_ = current_;
```

Comment:
`oldCurrent_` doesn't need to be a private variable, as it's lifespan is just this one function.